Q.1- Name and handle the exception occured in the following program:


a=3
 if a<4:
    a=a/(a-3)
     print(a)

ERROR:-ZeroDivisionError


while True:
    try:
	    a=3
	    if a<4:
		a=a/(a-3)
		print(a)
    except ZeroDivisionError :
     a=a/(a-1)
     print("ZeroDivisionError")
     print(a)
     break



'''Q.2- Name and handle the exception occurred in the following program:
l=[1,2,3]
print(l[3])'''


EXCEPTION:-IndexError

def function():
    try:
        print(l[3])

    except IndexError:
        print('index specified is wrong')
function()

#Q.3- What will be the output of the following code:
#Program to depict Raising Exception
 
try:
    raise NameError("Hi there")  # Raise Error
except NameError:
    print "An exception"
    raise  # To determine whether the exception was raised or not

OUTPUT:

An exception

Traceback (most recent call last):
  File "<pyshell#1>", line 2, in <module>
    raise NameError("Hi there")  # Raise Error
NameError: Hi there


 '''Q.4- What will be the output of the following code:


 # Function which returns a/b
def AbyB(a , b):
	try:
		c = ((a+b) / (a-b))
	except ZeroDivisionError:
		print "a/b result in 0"
	else:
		print c

'''Driver program to test above function
AbyB(2.0, 3.0)
AbyB(3.0, 3.0)'''

output:

AbyB(2.0, 3.0)
-5.0
AbyB(3.0, 3.0)
a/b result in 0




Q.5- Write a program to show and handle following exceptions:
1. Import Error
2. Value Error
3. Index Error


1.IMPORT ERROR:-

import xyz
def func():
        try:
            time.sleep(10)
        except ImportError:
            print('import right module')

func()



2.VALUE ERROR:-
def func():
        try:
            x=int(input("enter the  1st number"))
            y=int(input("enter the 2nd number"))
            sum=x+y
        except ValueError:
            print('enter integer only')

func()




3.INDEX ERROR
l=[1,2,3]
def function():
    try:
        print(l[3])

    except IndexError:
	     print('index specified is wrong')
function()


